c/tree.c

```diff
@@ -1145,8 +1153,15 @@ LOCAL void encode_node(MMDBW_tree_s *tree, MMDBW_node_s *node,
 
     uint32_t left =
         htonl(record_value_as_number(tree, &(node->left_record), args));
+    if (0 == left) {
```

Comment:
I wonder if there is a way to extend this check to other cycles. Is it true that a child node should always have a higher value than than a parent node? (Given how this implementation works, not a hypothetical implementation that follows the spec.)